python/shotgun_desktop/startup.py

```diff
@@ -313,6 +313,18 @@ def _quit(self):
         """
         self.exit(self.CLOSE_APP)
 
+    def exec_(self):
+        """
+        Execute the local event loop. If CmdQ was hit in the past, it will be handled just as if the
+        user had picked the Quit menu.
+        """
+        code = QtCore.QEventLoop.exec_(self)
+        # CmdQ was hit before.
+        if code == -1:
```

Comment:
In the QT docs, it says *By convention, a returnCode of 0 means success, and any non-zero value indicates an error.*

For completeness, since we return constants already, maybe we should enforce that no return values outside of constants are returned? So we could add a `ERROR` and `UNKNOWN`, where error catches -2 and below and unknown everything else?